src/probe.c

```diff
@@ -344,9 +344,9 @@ int main(int argc, char **argv)
                 remove_client(&(daemon.client_head), cl);
         }
         free(daemon.pollfds);
-
-        free(daemon.machine_id_override);
-
+        if (daemon.machine_id_override != NULL) {
```

Comment:
If the pointer is NULL, calling free() on it is a no-op. So, technically this change is not needed, but it's fine either way.